File: Guanabara - POO Java/Meus exercícios POO/Aula10/src/main/java/heranca/Funcionario.java
package heranca;
public class Funcionario extends Pessoa {//herança
    //Atributos
    private String setor;
    private boolean trabalhando;
    
    //Métodos Abstratos
    public void mudarTrabalho() {
        this.trabalhando = ! this.trabalhando; //inverte a lógica
    }
    //Métodos Acessores
    public String getSetor() {
        return setor;
    }

    public void setSetor(String setor) {
        this.setor = setor;
    }

    public boolean isTrabalhando() {
        return trabalhando;
    }

    public void setTrabalhando(boolean trabalhando) {
        this.trabalhando = trabalhando;
    }
}

File: Guanabara - PHP/exercicios-PHPOO/Aula10-HerancaPessoa/Funcionario.php
<?php
require_once "Pessoa.php";
class Funcionario extends Pessoa
{
    //Atributos
    private $setor;
    private $trabalhando;

    //Métodos Especias
    function mudarTrabalho($novoSetor)
    {
        $this->setor = $novoSetor;
        echo "<p>A funcionária <strong>" . $this->nome . "</strong> foi movida para o setor " . $novoSetor . ".</p>";
    }
    function demitirFuncinario()
    {
        $this->trabalhando = false;
        echo "<p>A funcionária <strong>" . $this->nome . "</strong> foi demitida.</p>";
    }

    //Métodos acessores
    function getSetor()
    {
        return $this->setor;
    }
    function getTrabalhando()
    {
        return $this->trabalhando;
    }
    function setSetor($Setor)
    {
        $this->setor = $Setor;
    }
    function setTrabalhando($boolean)
    {
        $this->trabalhando = $boolean;
    }
}
